### ISSUE TITLE
Console: cannot pass an argument with spaces

### ISSUE THREAD

**Issue by spyder-issues-migrator** [spyder-ide/spyder#1455]
_From [joonhyou...@gmail.com](https://code.google.com/u/118273923546112833546/) on 2013-06-23T20:56:46Z_

Please note that reporting bugs from Spyder's help menu ("?" > "Report
issue...") will fill version details for Spyder, Python and Qt in this description for you. 

What steps will reproduce the problem?

 1. In run settings (F6), for command line options, set an option with string with spaces, for example, first second or "first second"
2. Check Execute in current Python or Ipython interpreter
3. Run debug, with IPython interpreter open What is the expected output? What do you see instead? The argument is passed without an issue. Instead, I get

    Example.py: error: unrecognized arguments: second"

or

    Example.py: error: unrecognized arguments: second What version of the product are you using? On what operating system? Spyder 2.2.0 with Anaconda 64bit in Windows 8 64bit 

Please provide any additional information below

.

_Original issue: http://code.google.com/p/spyderlib/issues/detail?id=1455_

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2013-09-04T11:55:30Z_

Thanks for reporting. I'll try to solve it for 2.3

**Labels:** Cat-Console MS-v2.3

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-05-04T15:04:26Z_

**Labels:** -MS-v2.3 MS-v2.3.1

**Comment by spyder-issues-migrator** [comment]
_From [ccordoba12](https://code.google.com/u/ccordoba12/) on 2014-08-17T18:41:56Z_

**Summary:** Console: cannot pass an argument with spaces  \<span class="oldvalue"> (was: QTconsole: cannot pass an argument with spaces) \</span>  
  **Labels:** -MS-v2.3.1 MS-v2.4

**Comment by goanpeca** [comment]
Is this still relevant?

**Comment by ccordoba12** [comment]
Probably, I don't know.

**Comment by arnauddgbes** [comment]
Yes, this is still relevant, see below a code snippet to reproduce the issue:

```
import argparse
parser = argparse.ArgumentParser()
parser.add_argument( '--foo', nargs=1, help='A string argument' )
args = vars(parser.parse_args())
print( args )
```

The following script is run with the command line arguments '--foo "a space delimited string"'.
It works well when run in the current console, in a dedicated console, but fails when executed in an external system command, with the error:
```
usage: script.py [-h] [--foo FOO]
script.py: error: unrecognized arguments: space delimited string
```

* Spyder version: 5.2.1 None
* Python version: 3.9.7 64-bit
* Qt version: 5.12.9
* PyQt5 version: 5.12.3
* Operating System: Windows 10
